### DISCUSSION TITLE
Adding a team relationship

### DISCUSSION THREAD

**Discussion by juliangums** [spatie/laravel-permission#2518] [General]
It would be great to access a role or permissions team via a relationship. This would make it easier to work with Filament for example. There could also be a setting to specify the name of the relationship.

**Comment by drbyte** [comment]
Are you talking about authorizing users' sessions? With Filament, when you log someone into their team (likely via middleware) you can simply then also set this package's global team id so that all lookups for the user use that team.

**Reply by juliangums** [reply]
You are talking about doing something like this in a tenant-aware middleware?
```php
        $filament = Filament::getTenant()->id;
        $spatie = getPermissionsTeamId();
        if ($filament !== $spatie) {
            setPermissionsTeamId($filament);
        }
```

That is a Filament-specific issue. It's not what I was referring to. I just need a relationship to access my team through. The Filament use-case was just an example. When tenancy is activated, a Role has a team_id (or whatever key you specify in team_foreign_key). So a role (optionally) belongs to a team, but there isn't any way to retrieve this relationship through the model, but I think there should be. I have to extend it like this to retrieve `$role->team`. 
```php
<?php

namespace App\Models;

use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
    public function team()
    {
        return $this->belongsTo(Team::class);
    }
}
```

I'm just not sure why this isn't already in the model in some way, as it would be super useful. Ideally, this should be optionally added if `teams` is true and maybe there could be a variable in the config to change the name of the relationship and the class used. The foreign id is already specified in `team_foreign_key`.

Thoughts?

**Reply by drbyte** [reply]
Thanks for clarifying. 

I see your point.

**Comment by drbyte** [comment]
@erikn69 What's your approach been for this?

**Reply by erikn69** [reply]
>I have to extend it like this to retrieve $role->team

I have not needed to obtain the team through the role, I only get the roles I require by the team id with `where`. 

>The foreign id is already specified in team_foreign_key

The local key could be custom too

**Reply by juliangums** [reply]
It would definitely be useful though. That is for both roles and permissions, btw. There's some other people having the same problem here https://github.com/Althinect/filament-spatie-roles-permissions/issues/91 having to extend the model

**Comment by drbyte** [comment]
@juliangums I had hoped to have time to look into this, but that hasn't happened.

Do you want to take a stab at a PR? (with tests)

**Comment by juliangums** [comment]
@drbyte I made a start, but am pretty busy right now. I also suck with tests. Maybe have a look at this and let me know if this is what you expected and if yes I (or someone else) can have a look at writing the tests for it thereafter: https://github.com/spatie/laravel-permission/compare/main...juliangums:laravel-permission:patch-2
